server.js

```diff
@@ -20,4 +20,57 @@ app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
+app.get('/messages', function (request, response) {
+  response.json(messages)
+})
+
+app.post('/messages', function (request, response) {
+  let newMessage = {};
+
+  if (request.body.from.trim() && request.body.text.trim()) {
+    newMessage.id = messages.length
+    newMessage.from = request.body.from
+    newMessage.text = request.body.text
+    newMessage.timeSent = new Date()
+    messages.push(newMessage)
+    response.json(messages)
+  } else {
+    response.status(400).json("Please fill all fileds")
+  }
+
+})
+
+app.get('/messages/search', function (request, response) {
+  let searchWord = request.query.text
+  let result = messages.filter(item => item.text.toLowerCase().includes(searchWord.toLowerCase()))
+  if (result.length > 0) {
+    response.json(result)
+  } else {
+    response.status(404).json("Nothing was found")
+  }
+})
+
+app.get('/messages/latest', function (request, response) {
+  let latestMessages = messages.splice(-2)
```

Comment:
The assignment asks that you return the most recent 10 - but that is an easy fix!

Reply:
```suggestion
  let latestMessages = messages.slice(-10)
```
This should be better now.... I actually found this out 2 days back when I was trying to build the front end for this. and I realized that I used the wrong function.